src/Modal.js

```diff
@@ -277,6 +277,11 @@ const Modal = React.createClass({
     }
   },
 
+  componentWillUnmount() {
+    this.onHide();
+    this.props.onHide();
```

Comment:
Ya the modal should not call props.onHide, all that needs to happen is that the listeners get cleaned up which is done in this.onHide